syntax = "proto3";

message Message {
  string content = 1;
}

message GetMessagesRequest {
  int32 n = 1;
}

message BulkMessageRequest {
  repeated Message messages = 1;
}

message GetMessagesWithHashtagRequest {
  string hashtag = 1;
}

service MiniTwitter {
  rpc sendMessage(Message) returns (Message);
  rpc getMessages(GetMessagesRequest) returns (stream Message);
  rpc sendBulkMessages(BulkMessageRequest) returns (stream Message);
  rpc getMessagesWithHashtag(GetMessagesWithHashtagRequest) returns (stream Message);
}
